trieste/models/gpflow/models.py

```diff
@@ -57,8 +57,28 @@ def __init__(
             )
         self._num_kernel_samples = num_kernel_samples
 
+        self._ensure_variable_model_data()
+
+    def __repr__(self) -> str:
+        """"""
+        return f"GaussianProcessRegression({self._model!r}, {self.optimizer!r})"
+
+    @property
+    def model(self) -> GPR | SGPR:
+        return self._model
+
+    def _ensure_variable_model_data(self) -> None:
         # GPflow stores the data in Tensors. However, since we want to be able to update the data
         # without having to retrace the acquisition functions, put it in Variables instead.
+        # To tell if the data is stored in a way it can be updated without retracing,
+        # we look at its first dimension size.
+        # If it is a number, TF won't allow it to change, and we reset it to None
+
+        if self._model.data[0].shape[0] is None and self._model.data[1].shape[0] is None:
```

Comment:
Maybe as follows (slightly cleaner and more robust):

```suggestion
        if all(isinstance(x, tf.Variable) and x.shape[0] is None for x in self._model.data):
```

Reply:
looks good!